java-frontend/src/main/java/org/sonar/java/collections/AVLTree.java

```diff
@@ -156,8 +151,11 @@ public boolean hasNext() {
 
     private void descendLeft(AVLTree node) {
       if (node != EMPTY) {
-        stack.addLast(node);
         descendLeft(node.left());
+        while (node != null) {
+          stack.add(node);
+          node = node.nextInBucket();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Introduce a new variable instead of reusing the parameter "node". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1226)